backend/src/stwo/prover.rs

```diff
@@ -253,37 +267,20 @@ where
                         "All witness columns in a single machine must have the same length"
                     );
 
-                    if let Some(preprocessed) = self.proving_key.preprocessed.as_ref() {
-                        if let Some(table_provingkey) =
-                            preprocessed
-                                .iter()
-                                .find_map(|(namespace, table_provingkey)| {
-                                    if namespace == machine {
-                                        Some(table_provingkey)
-                                    } else {
-                                        None
-                                    }
-                                })
-                        {
-                            if let Some(constant_trace) =
-                                table_provingkey
-                                    .iter()
-                                    .find_map(|(size, table_provingkey)| {
-                                        if *size == machine_length {
-                                            Some(
-                                                table_provingkey
-                                                    .constant_trace_circle_domain
-                                                    .clone(),
-                                            )
-                                        } else {
-                                            None
-                                        }
-                                    })
-                            {
-                                // Extend constant_cols only if the constant_trace exists
-                                constant_cols.extend(constant_trace);
-                            }
-                        }
+                    if let Some(constant_trace) = self
```

Comment:
I suspect this can be removed as well, by calling `extend` directly on the option